### PR TITLE
Draft : Topology Deployer Web App

### PR DESCRIPTION
**Pull request by vpatel95**

### CODE REVIEW

File: web-app/ui/src/contexts/UserContext.js
Comment (vpatel95) [inline web-app/ui/src/contexts/UserContext.js:1]: STALE.

File: web-app/ui/src/wrapper/AuthWrapper.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Navigate, Outlet } from "react-router-dom";
+import {SessionStore} from "services/store";
+
+export const AuthWrapper = () => {
+  const user = SessionStore.getUser();
+
```
Comment (iamprakharjaiswal) [inline web-app/ui/src/wrapper/AuthWrapper.js]: yahan pe sessionstore ki jagah auth context use kar skta

Comment (vpatel95) [reply web-app/ui/src/wrapper/AuthWrapper.js]: Hn lekin, auth context is not accessible in loader where I need to fetch current user objects using id from the user

File: web-app/ui/src/services/api.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+const BASE_URL = 'http://10.87.1.25:5000/api';
+
+const Axios = axios.create({
+  baseURL: BASE_URL,
+  withCredentials: true,
+});
+
+Axios.interceptors.request.use(
+  config => {
```
Comment (iamprakharjaiswal) [inline web-app/ui/src/services/api.js]: auth wrapper me check karle if user is authenticated or not, if not call this method and authenticate the user, ideally api methods should only call apis and return the response, they should not set the state

Comment (vpatel95) [reply web-app/ui/src/services/api.js]: getToken hai -_-

**Pull request closed without merging**